src/evaluate.jl

```diff
@@ -30,18 +31,41 @@ function evaluate(a::Taylor1{T}, dx::S) where {T<:Number, S<:Number}
 end
 evaluate(a::Taylor1{T}) where {T<:Number} = a[0]
 
+function evaluate(a::STaylor1{N,T}, dx::T) where {N, T<:Number}
```

Comment:
Maybe some functions like this can be defined only once for both `Taylor1` and `STaylor1`. 
Is there an `AbstractTaylor1` type?

Reply:
I've looked into this a little bit. I haven't added an AbstractTaylor1 type yet. 

We could do this for the evaluate(a::Taylor1{T}) function (just have both functions call get_order) but I haven't been able to figure out a way to avoid this for most other functions.